Source/Tools/LinuxEmulation/VDSO_Emulation.cpp

```diff
@@ -607,52 +607,66 @@ static std::array<FEXCore::IR::ThunkDefinition, 7> VDSODefinitions = {{
   },
 }};
 
-void LoadGuestVDSOSymbols(bool Is64Bit, char* VDSOBase) {
+template<bool Is64Bit>
+void LoadGuestVDSOSymbols(char* VDSOBase) {
+  using ELFHeaderType = std::conditional_t<Is64Bit, Elf64_Ehdr, Elf32_Ehdr>;
+  using ELFSHeaderType = std::conditional_t<Is64Bit, Elf64_Shdr, Elf32_Shdr>;
+  using ELFSymbolType = std::conditional_t<Is64Bit, Elf64_Sym, Elf32_Sym>;
+
   // We need to load symbols we care about.
   if (Is64Bit) {
     // We don't care about any 64-bit symbols right now.
     return;
   }
 
   // 32-bit symbol loading.
-  const Elf32_Ehdr* Header = reinterpret_cast<const Elf32_Ehdr*>(VDSOBase);
+  auto Header = reinterpret_cast<const ELFHeaderType*>(VDSOBase);
 
   // First walk the section headers to find the symbol table.
-  const Elf32_Shdr* RawShdrs = reinterpret_cast<const Elf32_Shdr*>(VDSOBase + Header->e_shoff);
+  auto RawShdrs = reinterpret_cast<const ELFSHeaderType*>(VDSOBase + Header->e_shoff);
 
-  const Elf32_Shdr* StrHeader = &RawShdrs[Header->e_shstrndx];
+  const auto StrHeader = &RawShdrs[Header->e_shstrndx];
   const char* SHStrings = VDSOBase + StrHeader->sh_offset;
 
-  const Elf32_Shdr* SymTableHeader {};
-  const Elf32_Shdr* StringTableHeader {};
+  const ELFSHeaderType* SymTableHeader {};
+  const ELFSHeaderType* StringTableHeader {};
 
-  for (size_t i = 0; i < Header->e_shnum; ++i) {
-    const auto& Header = RawShdrs[i];
-    if (Header.sh_type == SHT_SYMTAB && strcmp(&SHStrings[Header.sh_name], ".symtab") == 0) {
-      SymTableHeader = &Header;
-      StringTableHeader = &RawShdrs[SymTableHeader->sh_link];
-      break;
+  struct SymbolTypes {
+    const char* name;
+    int sh_type;
+  };
+
+  constexpr std::array<SymbolTypes, 2> symbol_table_names = {{{".dynsym", SHT_DYNSYM}, {".symtab", SHT_SYMTAB}}};
+
+  for (auto sym_table : symbol_table_names) {
```

Comment:
I would've gone for `std::pair` over a named struct and then using `auto [name, sh_type]` here. Can leave it as is though, just mentioning in case it's interesting.